domain/resource/state/types.go

```diff
@@ -108,9 +108,13 @@ type kubernetesApplicationResource struct {
 type storedFileResource struct {
 	ObjectStoreUUID string `db:"store_uuid"`
 	ResourceUUID    string `db:"resource_uuid"`
+	Size            int64  `db:"size"`
+	Fingerprint     string `db:"fingerprint"`
 }
 
 type storedContainerImageResource struct {
 	StorageKey   string `db:"store_storage_key"`
 	ResourceUUID string `db:"resource_uuid"`
+	Size         int64  `db:"size"`
+	Fingerprint  string `db:"fingerprint"`
```

Comment:
This should be named hash plain and simple and it should have the encoding type either associated with it or have it named in the column.

See charm_hash.

Reply:
I've made this change (encoding named in the column), and used "hash" rather than fingerprint in the state layer but left the service layer as is because the resource store uses the fingerprint. This could be changed though if it would be better to just use hash everywhere and try and do away with the fingerprint concept.